consensus/src/epoch_manager.rs

```diff
@@ -275,6 +294,98 @@ impl EpochManager {
         Ok(())
     }
 
+    fn prepare_decoupled_execution(
+        &mut self,
+        epoch: u64,
+        recovery_data: RecoveryData,
+        epoch_state: &EpochState,
+        safety_rules_container: Arc<Mutex<MetricsSafetyRules>>,
+        network_sender: &NetworkSender,
+    ) -> (
+        ProposalGenerator,
+        ExecutionPhase,
+        CommitPhase,
+        Arc<BlockStore>,
+    ) {
+        let guage_e = IntGauge::new(
+            "D_EXE_CHANNEL_COUNTER",
```

Comment:
the name doesn't make much sense to me, also we should register it in counter.rs

Reply:
resolved